src/wms/src/main/java/org/geoserver/wms/map/DirectRasterRenderer.java

```diff
@@ -1194,5 +1190,21 @@ protected GridCoverage2D readCoverage(
                 throw new IOException("Failure rendering the coverage", e);
             }
         }
+
+        private GeneralParameterValue[] setInterpolation(
+                Interpolation interpolation, GeneralParameterValue[] readParams) {
+            if (interpolation != null) {
+                List<GeneralParameterValue> paramList = new ArrayList<>();
+                if (readParams != null) {
+                    paramList.addAll(Arrays.asList(readParams));
+                }
+                final Parameter<Interpolation> readInterpolation =
+                        (Parameter<Interpolation>) ImageMosaicFormat.INTERPOLATION.createValue();
+                readInterpolation.setValue(interpolation);
+                paramList.add(readInterpolation);
```

Comment:
What if an interpolation read parameter was already in the mix? I guess it would have to be replaced instead?

Reply:
Feedback has been addressed.